plugin_test.go

```diff
@@ -0,0 +1,58 @@
+package plugin
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestInvalidConfig(t *testing.T) {
+	cfg := CreateConfig()
+	next := http.HandlerFunc(func(http.ResponseWriter, *http.Request) {})
+	_, err := New(context.Background(), next, cfg, "traefik-tls-headers-plugin")
+	if err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func TestTLSCipher(t *testing.T) {
+	cfg := CreateConfig()
+	cfg.Headers.Cipher = "X-TLS-Cipher"
+	next := http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+		assertHeader(t, r.Header, "X-TLS-Cipher", "TLS_AES_128_GCM_SHA256")
```

Comment:
nit: canonicalize the header name

```suggestion
	cfg.Headers.Cipher = "X-Tls-Cipher"
	next := http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
		assertHeader(t, r.Header, "X-Tls-Cipher", "TLS_AES_128_GCM_SHA256")
```